vernac/declare.ml

```diff
@@ -454,6 +442,31 @@ let declare_constant_core ~name ~typing_flags cd =
   if unsafe || is_unsafe_typing_flags typing_flags then feedback_axiom();
   kn, delayed
 
+let declare_primitive_core ?typing_flags ~name e =
+  let typ, ubinders, ctx = match e.prim_entry_type with
+    | None -> None, UnivNames.empty_binders, Univ.ContextSet.empty
+    | Some (typ, (univs, ubinders)) ->
+      let univ_entry, ctx = extract_monomorphic univs in
+      Some (typ, univ_entry), ubinders, ctx
+  in
+  let () = Global.push_context_set ~strict:true ctx in
+  let e = {
+    Entries.prim_entry_type = typ;
+    Entries.prim_entry_content = e.prim_entry_content;
+  } in
+  let ubinders = (UState.Monomorphic_entry ctx, ubinders) in
+  let decl = ConstantEntry (Entries.PrimitiveEntry e) in
+  let kn = Global.add_constant ?typing_flags name decl in
+  let () = DeclareUniv.declare_univ_binders (GlobRef.ConstRef kn) ubinders in
+  if is_unsafe_typing_flags typing_flags then feedback_axiom();
+  kn
+
+let declare_primitive ?(local = Locality.ImportDefaultBehavior) ~name ?typing_flags ?user_warns cd =
+  let () = check_exists name in
+  let kn = declare_primitive_core ?typing_flags ~name cd in
+  let () = register_constant kn IsPrimitive local ?user_warns in
+  kn
+
```

Comment:
By curiosity, why to extract a specific `declare_primitive`? It is because primitive cannot be polymorphic?

Reply:
See the message on the commit which does this change

Reply:
OK.

One reason I'm asking is that I started to be afraid at some time by the diversity of code doing this or that manipulation on universes which is sometimes the same sometimes different of what is done in similar functions, and, at the end I felt unable to evaluate whether it is normal that there are difference or if there were oversights.

For instance, so that it is easier to be convinced that the treatment done in `declare_primitive_core` is consistent with what is done in `declare_constant_core` would it make sense to encapsulate the following lines beyond a level of abstraction shared with `declare_constant_core`:
```ocaml
let typ, ubinders, ctx = ... in
let () = Global.push_context_set ~strict:true ctx in
let ubinders = (UState.Monomorphic_entry ctx, ubinders) in
let () = DeclareUniv.declare_univ_binders (GlobRef.ConstRef kn) ubinders in
```
so that it is clear that it is intended to work as in `declare_constant_core`?

For instance, are `Global.push_context_set` and `DeclareUniv.declare_univ_binders` always supposed to go in pair?

Or, would it make sense to change the level of abstraction of `extract_monomorphic` to take a pair `(univs, ubinders)` and return a triple `(univ_entry, ctx, (UState.Monomorphic_entry ctx, ubinders))`?

PS: My question is more general than the PR and it is not blocking for the PR.